src/controllers/AbstractController.ts

```diff
@@ -180,13 +182,32 @@ export default abstract class AbstractController<T extends AbstractService> {
 	protected parseNumberOrThrow(n: string, errorMessage: string): number {
 		const num = Number(n);
 
-		if (!Number.isInteger(num) || num < 0) {
+		if (!this.verifyInt(num)) {
 			throw new BadRequest(errorMessage);
 		}
 
 		return num;
 	}
 
+	/**
+	 * Expected format ie: 0-999
+	 */
+	protected parseRangeOfNumbersOrThrow(n: string): number[] {
+		const splitRange = n.split('-');
+		if (splitRange.length !== 2) {
+			throw new BadRequest('Incorrect range format. Expected example: 0-999');
+		}
+
+		const min = Number(splitRange[0]);
+		const max = Number(splitRange[1]);
+
+		if (!this.verifyInt(min) || !this.verifyInt(max)) {
+			throw new BadRequest('Inputted range contains non integers.');
+		}
+
+		return [...Array(max - min + 1).keys()].map((i) => i + min);
```

Comment:
I think you should check that `max >= min`

Reply:
If somebody asks for the range 1-10000000, this sort of thing will blow up quite quickly I guess?

It might be better to represent the range like some ` class Range { start: number, end: number }` that can be turned into an iterator to iterate over each number?

Reply:
So I did actually think of this. But not sure what the limit should be. On a local archive node I was able to query 10,000 blocks within 30 seconds. So my guess is to put a 10k max on it, with an overrideMax query param?

Reply:
if there was an `overrideMax` query param, wouldn't that allowa request to blow up the server still? Or do you mean ENV var perhaps?

How many blocks is something likely to realistically want to act for at once, I wonder? A response containing 10k blocks would still be quite large and be quite an easy DoS vector for any public sidecar instance (though that might be true of other requests, and if sidecar is mainly used internally maybe it's not a huge deal?)

I'd go for something like a 100-200ish max offhand, without knowing what people actually would be requesting back?

Reply:
Yea I think in general, users shouldn't ever really be using sidecar against public nodes and just their own archive nodes. Or atleast thats what all the exchanges, external teams use it for. I would propose maybe a 500 max with no override at all. Considering the requests are generally pretty fast all the way up to 1k blocks.

Reply:
I'd be up for 500! I'd also have no issue with an env var or other config parameter so that exchanges etc can customise it if they want.

Reply:
fixed